test/unit/handlers/authorizations/{ID}.test.js

```diff
@@ -112,72 +110,63 @@ describe('/authorizations/{ID}', () => {
       // Assert
       expect(response.statusCode).toBe(200)
       expect(Handler.forwardAuthorizationMessage).toHaveBeenCalledTimes(1)
-      expect(Handler.forwardAuthorizationMessage.mock.calls[0][2]).toEqual(options.payload)
-      expect(Handler.forwardAuthorizationMessage.mock.calls[0][3]).toEqual('PUT')
+      expect(Handler.forwardAuthorizationMessage.mock.calls[0][2]).toEqual(request.body)
+      expect(Handler.forwardAuthorizationMessage.mock.calls[0][3]).toEqual(method.toUpperCase())
     })
 
-    it('properly handles QRCODE payload', async () => {
+    it('handles when an error is thrown', async () => {
+      const request = await Mockgen.generateRequest(path, method)
+
       // Arrange
-      const mock = await requests
       const options = {
-        method: 'put',
-        url: '' + mock.request.path,
-        headers: Helper.defaultHeaders(),
-        payload: {
-          responseType: 'ENTERED',
-          authenticationInfo: {
-            authentication: 'QRCODE',
-            authenticationValue: 'abcdefghijklmnopqrstuvwxyz0987654321'
-          }
-        }
+        method,
+        url: path,
+        headers: request.headers,
+        payload: request.body
       }
 
+      const err = new Error('Error occurred')
+      Handler.forwardAuthorizationMessage.mockImplementation(() => { throw err })
+
       // Act
       const response = await server.inject(options)
 
       // Assert
-      expect(response.statusCode).toBe(200)
-      expect(Handler.forwardAuthorizationMessage).toHaveBeenCalledTimes(1)
-      expect(Handler.forwardAuthorizationMessage.mock.calls[0][2]).toEqual(options.payload)
-      expect(Handler.forwardAuthorizationMessage.mock.calls[0][3]).toEqual('PUT')
+      expect(response.statusCode).toBe(500)
```

Comment:
here we are getting randomly unexpected 400

IMHO: this is because you are using Mocken.generateRequest in L118, and sometimes it gets in troubles with this additional validation logic: https://github.com/mojaloop/transaction-requests-service/blob/pisp/master/src/handlers/authorizations/%7BID%7D.js#L116-L126

Reply:
Ok. I updated the openapi file and added a snippet in mockgen to enforce that U2F is accomponied by an appropriate object.

Makes me think about our schema that AuthenticationType and AuthenticationValue be coupled at a higher level.
Instead of this where Type and Value are not coupled :thinking: but maybe thats a thought for another time and affects much wider scope than this ticket.
```
    AuthenticationInfo:
      title: AuthenticationInfo
      description: Data model for the complex type AuthenticationInfo
      required:
        - authentication
        - authenticationValue
      type: object
      properties:
        authentication:
          $ref: "#/components/schemas/AuthenticationType"
        authenticationValue:
          $ref: "#/components/schemas/AuthenticationValue"
```

something like this

```
    AuthenticationInfo:
      title: AuthenticationInfo
      description: Data model for the complex type AuthenticationInfo
      required:
        - authentication
        - authenticationValue
      type: object
      oneOf:
        - $ref: '#/components/schemas/OTPAuthenicationTypeAndValue'
        - $ref: '#/components/schemas/QRCODEAuthenicationTypeAndValue'
        - $ref: '#/components/schemas/U2FAuthenicationTypeAndValue'
```

Reply:
@kleyow I like your proposition and I agree it should be handled in separate story - could you create one?